daemon/cmd/fqdn.go

```diff
@@ -111,14 +111,15 @@ func (d *Daemon) bootstrapFQDN(possibleEndpoints map[uint16]*endpoint.Endpoint,
 		return fmt.Errorf("could not initialize regex LRU cache: %w", err)
 	}
 	dnsProxyConfig := dnsproxy.DNSProxyConfig{
-		Address:                "",
-		Port:                   port,
-		IPv4:                   option.Config.EnableIPv4,
-		IPv6:                   option.Config.EnableIPv6,
-		EnableDNSCompression:   option.Config.ToFQDNsEnableDNSCompression,
-		MaxRestoreDNSIPs:       option.Config.DNSMaxIPsPerRestoredRule,
-		ConcurrencyLimit:       option.Config.DNSProxyConcurrencyLimit,
-		ConcurrencyGracePeriod: option.Config.DNSProxyConcurrencyProcessingGracePeriod,
+		Address:                     "",
+		Port:                        port,
+		IPv4:                        option.Config.EnableIPv4,
+		IPv6:                        option.Config.EnableIPv6,
+		EnableDNSCompression:        option.Config.ToFQDNsEnableDNSCompression,
+		MaxRestoreDNSIPs:            option.Config.DNSMaxIPsPerRestoredRule,
+		ConcurrencyLimit:            option.Config.DNSProxyConcurrencyLimit,
+		ConcurrencyGracePeriod:      option.Config.DNSProxyConcurrencyProcessingGracePeriod,
+		AllowNonTransparentFallback: !option.Config.EncryptionEnabled(),
```

Comment:
What do you think of making `AllowNonTransparentFallback` an explicit option that we can have be default off (or on?) and allow the user to opt in to this specific behavior so that it is very clear what's happening and under what cluster condition/operation? We can then document what is means for this to be on/off so that the user can make the tradeoff explicitly themselves.